tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -840,11 +840,34 @@ export class WebGPUBackend extends KernelBackend {
     });
     this.ensureCommandEncoderReady();
     const passEncoder = this.currentCommandEncoder.beginComputePass();
+    const shouldTimeProgram = this.activeTimers != null;
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        passEncoder.writeTimestamp(this.querySet, 0);
+      }
+    }
     passEncoder.setPipeline(program.pipeline);
     passEncoder.setBindGroup(0, bindGroup);
     passEncoder.dispatch(
         program.dispatch[0], program.dispatch[1], program.dispatch[2]);
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        passEncoder.writeTimestamp(this.querySet, 1);
+      }
+    }
     passEncoder.endPass();
+    this.computePassNumberInEncoder++;
+    this.commandQueueOwnedIds.add(outputId);
+    if (env().get('WEBGPU_DEFERRED_SUBMIT_BATCH_SIZE') as
+        number <= this.computePassNumberInEncoder) {
+      this.submitQueue();
```

Comment:
Sorry, one more comment: we should remove the threshold WEBGPU_DEFERRED_SUBMIT_BATCH_SIZE checking for fromPixels in this PR. Currently the texture is managed by the fromPixelsProgram. If the texture size changes, the old texture will be destroyed, which results the error that you are using a destroyed texture when submit if you are executing multiple fromPixels sequentially. You can find the warnings when executing `yarn test`. In the next step, you can consider how to support `WEBGPU_DEFERRED_SUBMIT_BATCH_SIZE` for fromPixels.

Reply:
Updated, PTAL